### PR TITLE
#18072 : Improving logging code in order to provide better error mess…

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…ages when a reindex fails. A more descriptive message has also been changed in the ESContentFactoryImpl class because of the upgrade in our ES library, which might involve updating configuration parameters and mappings for ES and existing contents in a customer environment.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 1]
```diff
@@ -277,20 +249,72 @@ public void setLanguageId(long languageId) {
...
+     * Returns the Content Type that this Contentlet belongs to.
+     *
+     * @return The {@link ContentType} object.
+     */
+    public ContentType getContentType() {
+        final ContentType contentType;
+        try {
+            contentType = APILocator.getContentTypeAPI(APILocator.systemUser()).find(contentTypeId);
+            return contentType;
+        } catch (final DotSecurityException e) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: better:

```suggestion
        } catch (final DotSecurityException | DotDataException e) {
          ...
```

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 2]
```diff
@@ -1599,159 +1619,109 @@ public void setDisabledWysiwyg(String disabledWysiwyg) {
...
-	 */
-	public void setField(String fieldVarName, Object value) throws DotRuntimeException {
-		Structure st = CacheLocator.getContentTypeCache().getStructureByInode(this.structureInode);
-		Field f = st.getFieldVar(fieldVarName);
-		if(f == null)
-			throw new DotRuntimeException("Field: " + fieldVarName + " doesn't exist.");
-		setField(f, value);
-	}
+			BeanUtils.setProperty(this, field.getFieldContentlet(), value);
+        } catch (final IllegalArgumentException iae) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: better:
```suggestion
        } catch (final DotRuntimeException | InvocationTargetException  | IllegalAccessException  e) {

```

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 3]
```diff
@@ -219,36 +221,6 @@
...
-    private File binary16;
-    private File binary17;
-    private File binary18;
-    private File binary19;
-    private File binary20;
-    private File binary21;
-    private File binary22;
-    private File binary23;
-    private File binary24;
-    private File binary25;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: why all this are remove?

**Review by freddyucv — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java [hunk 4]
```diff
@@ -1584,20 +1585,18 @@ protected SearchHits indexSearch(String query, int limit, int offset, String sor
...
-
-        } catch (ElasticsearchStatusException | IndexNotFoundException | SearchPhaseExecutionException infe ) {
-            Logger.warn(this.getClass(), "----------------------------------------------");
-            Logger.warn(this.getClass(), "Elasticsearch Index Error : " + indexToHit);
-            Logger.warnAndDebug(this.getClass(), infe.getMessage(), infe);
-            Logger.warn(this.getClass(), "----------------------------------------------");
-
+        } catch (final ElasticsearchStatusException | IndexNotFoundException | SearchPhaseExecutionException e) {
+            Logger.error(this.getClass(), "----------------------------------------------");
+            Logger.error(this.getClass(), String.format("Elasticsearch error in index '%s'", indexToHit));
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: are you sure that you wanna show these exceptions as error? they might cause a lot of noise. I think Logger.warnAndDebug is better

Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java]: Fixed

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java (hunk 3 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: That code is marked by IntelliJ as not being used by any other class in the source code. It was probable caused by another refactoring not related to this issue.

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java (hunk 1 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: Done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java (hunk 2 above)
Comment (jcastro-dotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: Done

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 5]
```diff
@@ -277,20 +249,72 @@ public void setLanguageId(long languageId) {
...
+    }
+
     /**
+     * @deprecated Use the method {@link #getContentType()}
+     *
+     * @return
      */
     public Structure getStructure() {
-        Structure structure = CacheLocator.getContentTypeCache().getStructureByInode(structureInode);
+        Structure structure = CacheLocator.getContentTypeCache().getStructureByInode(contentTypeId);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: Codacy Issue found: [Local variable 'structure' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5120047)

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 6]
```diff
@@ -277,20 +249,72 @@ public void setLanguageId(long languageId) {
...
+     */
+    public ContentType getContentType() {
+        final ContentType contentType;
+        try {
+            contentType = APILocator.getContentTypeAPI(APILocator.systemUser()).find(contentTypeId);
+            return contentType;
+        } catch (final DotSecurityException e) {
+            Logger.error(this, String.format("An error occurred when returning the Content Type associated to " +
+                    "Contentlet '%s'", this.getIdentifier()));
+        } catch (final DotDataException e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: Codacy Issue found: ['catch' branch identical to 'DotSecurityException' branch](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5120047)

**Review by freddyucv — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java [hunk 7]
```diff
@@ -277,20 +249,69 @@ public void setLanguageId(long languageId) {
...
+    }
+
     /**
+     * @deprecated Use the method {@link #getContentType()}
+     *
+     * @return
      */
     public Structure getStructure() {
-        Structure structure = CacheLocator.getContentTypeCache().getStructureByInode(structureInode);
+        Structure structure = CacheLocator.getContentTypeCache().getStructureByInode(contentTypeId);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/Contentlet.java]: Codacy Issue found: [Local variable 'structure' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5120047)

**Review by dsilvam — APPROVED** [review]

**Pull request merged**